### ISSUE TITLE
non-OK status: GpuLaunchKernel - Internal: no kernel image is available for execution on the device

### ISSUE THREAD

**Issue by allanchao** [SciSharp/TensorFlow.NET#685]
I'm experiencing a TensorFlow.net GPU where CUDA runs fine on sample projects but does not work for machine learning examples.

Here is the full error:

**Non-OK-status: GpuLaunchKernel(FillPhiloxRandomKernelLaunch, num_blocks, block_size, 0, d.stream(), gen, data, size, dist) status: Internal: no kernel image is available for execution on the device"**

**Specs**
Visual Studio 2019 Community and Preview (same problem on both)
GeForce 860M (compute capability 5.0)
Cuda 10.1 (cuda_10.1.243_426.00_windows)
Cudnn 7.6.5 (cudnn-10.1-windows7-x64-v7.6.5.32.zip)
Windows 8.1

I read that [Tensorflow 2.3 requires a higher compute capability](https://stackoverflow.com/questions/63715824/no-kernel-image-is-available-for-execution-on-the-device-fatal-python-error-abo), but I'm not sure if that's the real cause. If so, is there any workaround have a working Gpu-Tensorflow for older graphics cards still on compute capability 5.0?

I have already tested and ensured that cudnn works properly [using the sample projects](https://docs.nvidia.com/deeplearning/cudnn/install-guide/index.html#installwindows).

![image](https://user-images.githubusercontent.com/1066566/102162566-3d2a2a00-3e3e-11eb-8e6a-2cd555a33cfe.png)

**Comment by Esther2013** [comment]
Same as https://github.com/SciSharp/TensorFlow.NET/issues/617

**Comment by allanchao** [comment]
> Same as #617

Yeah 617 is closed so I wanted to report it on an issue that's open

**Comment by allanchao** [comment]
OK I discovered and resolved the problem on my computer. I'll post the solution here in case it helps anyone.

My problem was that when following the instructions to use nuget package manager to remove SciSharp.TensorFlow.Redist and replace it with SciSharp.TensorFlow.Redist-Windows-GPU, I had blindly selected the latest version.  Instead, I needed to choose the version that matched the other one. Originally I had changed from 1.15.1 --> 2.4.0 (latest), but that's what caused the problem. I downgraded to 1.15.1 and cleaned and ran it again, and it worked this time.
